py/runai_model_streamer/runai_model_streamer/file_streamer/file_streamer.py

```diff
@@ -80,10 +84,17 @@ def stream_files(
             self,
             file_stream_requests: List[FileChunks],
             credentials: Optional[S3Credentials] = None,
+            device: Optional[str] = None,
 ) -> None:
         if not homogeneous_paths([file_stream_request.path for file_stream_request in file_stream_requests]):
             raise RunaiStreamerInvalidInputException("Cannot stream files from multiple source types in parallel") 
 
+        if device is None:
+            self.device_str = "cpu"
+        else:
+            self.device_str = device
+        self.device_type = torch.device(self.device_str)
```

Comment:
"self.device_type" gives the hint its just the type, but you store in it the device object. 
So maybe "self.device" is better name.

What do you think?